### PR TITLE
Upgrade pylint

### PR DESCRIPTION
**Pull request by cpennington**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -20,9 +20,11 @@ classifiers =
 [options]
 packages = find:
 install_requires =
-    pylint==1.7.6
-    pylint-django==0.7.2
-    pylint-celery==0.3
+    pylint==1.9.5;python_version<"3.0"  # The last pylint version to support python 2
+    pylint==2.4.2;python_version>="3.0"
```
Comment (nedbat) [inline setup.cfg]: Do we run the risk of warnings vacillating as the same repo is linted sometimes with python 2, sometimes with python 3?

File: setup.cfg [hunk 2]
```diff
@@ -20,9 +20,11 @@ classifiers =
...
+    pylint-django==0.11.1;python_version<"3.0"  # The last pylint-django to support python 2
+    pylint==2.0.11;python_version>="3.0"
```
Comment (nedbat) [inline setup.cfg]: This line is a duplicate of line 24, but with a different version?

File: setup.cfg [hunk 3]
```diff
@@ -20,9 +20,11 @@ classifiers =
...
+    pylint-celery
```
Comment (nedbat) [inline setup.cfg]: We don't want to pin this package?

File: setup.cfg (hunk 2 above)
Comment (cpennington) [reply setup.cfg]: oops. supposed to be `pylint-django`

File: setup.cfg (hunk 1 above)
Comment (cpennington) [reply setup.cfg]: Maybe? But I think our linting jobs should just be pinned to either lint w/ 2, or lint w/ 3, but not both.

**Pull request merged**